### DISCUSSION TITLE
Optionally, delay the visual appearance of fly pie for quick activation?

### DISCUSSION THREAD

**Discussion by Leleat** [Schneegans/Fly-Pie#126] [Ideas & Feature Discussions]
I'd like to use fly-pie in a pretty basic way (only for window management) and only go 1 layer deep. So I think having an option to delay the visual appearance of fly pie (similiar to how GNOME's altTab delays its appearance) would be good since with just 1 layer I know where all the options are and that brief flashing of the fly pie is more disruptive than helpful.

**Comment by Schneegans** [comment]
That's actually a pretty good idea! Would it be something you would like to specify globally, for all menus or on a per-menu-basis?

**Reply by Leleat** [reply]
Personally, a global setting would be enough.

Maybe as a middle ground between simple menu's where all items are immediately obvious where they are, you could abort the delay / show the menu immediately, if the user entered the 2nd or 3rd layer of the pie menu.
(Just like the altTab popup immediately shows up, if you press `Tab` twice.)

But then again, I don't use multiple / complex menus, so I don't know what other people would prefer.

**Comment by Schneegans** [comment]
An alternative approach could be a global option which keeps the menu hidden until the mouse pointer does not move for a couple of milliseconds. So the menu will stay hidden until you stop moving your mouse! What do you think?

**Reply by Leleat** [reply]
Sure, sounds like a good approach as well. Only "theoretical" downside I can think of, is that if people use 3+ layers, they may need to wait to see what they will actually activate with the gesture and go back *multiple* layers, if they were at the wrong item. This may feel "like lag".

But I am not sure how much of an issue that would be in reality... a) how many people use that many layers and b) maybe people won't actually wait since their muscle memory may be precise enough.

Either approach works for me since I belong to a).

**Comment by Leleat** [comment]
So are you in favor of this / can I open an issue for it?

**Reply by Schneegans** [reply]
Yes!

I transferred this discussion to an issue: #147 

Let's continue the discussion over there!

**Comment by Schneegans** [comment]
I submitted an initial implementation of this feature to the `develop` branch. You can test it by [installing the current development version of Fly-Pie](https://github.com/Schneegans/Fly-Pie/blob/develop/docs/installation.md#c-cloning-the-latest-version-with-git)! (Feel free to ask is anything is unclear in this guide). Then:
* Open the settings of Fly-Pie.
* Go to "Advanced Settings".
* Set "Display Timeout" to something higher than zero. 150 works quite well for me.

What do you think?

**Reply by Leleat** [reply]
Thanks, I'll check it out tomorrow and report back.

**Reply by Leleat** [reply]
Got the chance to try it out and it works as expeced. Thanks!
